app/components/edit_channel_info/index.js

```diff
@@ -0,0 +1,422 @@
+        if (goBack) {
+            this.props.navigator.pop({animated: true});
+        } else {
+            this.props.navigator.dismissModal({
+                animationType: 'slide-down'
+            });
+        }
+    };
+
+    lastTextRef = (ref) => {
+        this.lastText = ref;
+    };
+
+    canUpdate = (displayName, channelURL, purpose, header) => {
+        const {
+            oldDisplayName,
+            oldChannelURL,
+            oldPurpose,
+            oldHeader
+        } = this.props;
+
+        return displayName !== oldDisplayName || channelURL !== oldChannelURL ||
+            purpose !== oldPurpose || header !== oldHeader;
+    };
+
+    enableRightButton = (enable = false) => {
+        this.props.enableRightButton(enable);
+    };
+
+    onDisplayNameChangeText = (displayName) => {
+        const {editing, onDisplayNameChange} = this.props;
+        onDisplayNameChange(displayName);
+        if (!editing && displayName && displayName.length >= 2) {
+            this.props.enableRightButton(true);
+        } else {
+            this.props.enableRightButton(false);
+        }
+
+        if (editing) {
+            const {channelURL, purpose, header} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onDisplayURLChangeText = (channelURL) => {
+        const {editing, onChannelURLChange} = this.props;
+        onChannelURLChange(channelURL);
+
+        if (editing) {
+            const {displayName, purpose, header} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onPurposeChangeText = (purpose) => {
+        const {editing, onPurposeChange} = this.props;
+        onPurposeChange(purpose);
+
+        if (editing) {
+            const {displayName, channelURL, header} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    onHeaderChangeText = (header) => {
+        const {editing, onHeaderChange} = this.props;
+        onHeaderChange(header);
+
+        if (editing) {
+            const {displayName, channelURL, purpose} = this.props;
+            const canUpdate = this.canUpdate(displayName, channelURL, purpose, header);
+            this.enableRightButton(canUpdate);
+        }
+    };
+
+    scrollRef = (ref) => {
+        this.scroll = ref;
+    };
+
+    scrollToEnd = () => {
+        this.scroll.scrollToFocusedInput(findNodeHandle(this.lastText));
+    };
+
+    render() {
+        const {theme, editing, channelType, currentTeamUrl} = this.props;
+        const {displayName, channelURL, header, purpose} = this.props;
+        const {error, saving} = this.props;
+        const {height, width} = Dimensions.get('window');
```

Comment:
you should be using the `deviceWidth` and `deviceHeight` available in the redux store